api/src/modules/scenarios/scenario.entity.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  BaseEntity,
+  OneToMany,
+  ManyToOne,
+} from 'typeorm';
+
+import { SourcingLocation } from 'modules/sourcing-locations/sourcing-location.entity';
+
+import { BaseServiceResource } from 'types/resource.interface';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { User } from '../users/user.entity';
+
+export enum SCENARIO_STATUS {
+  ACTIVE = 'active',
+  INACTIVE = 'inactive',
+  DELETED = 'deleted',
+}
+
+export const scenarioResource: BaseServiceResource = {
+  className: 'Scenario',
+  name: {
+    singular: 'scenario',
+    plural: 'scenarios',
+  },
+  entitiesAllowedAsIncludes: [],
+};
+
+@Entity()
+export class Scenario extends BaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  @ApiProperty()
+  id: string;
+
+  @ApiProperty()
+  @Column({ nullable: false })
+  title!: string;
+
+  @ApiPropertyOptional()
+  @Column({ nullable: true })
+  description?: string;
+
+  @ApiProperty()
+  @Column({
+    type: 'enum',
+    enum: SCENARIO_STATUS,
+    enumName: 'entityStatus',
+    default: SCENARIO_STATUS.INACTIVE,
+  })
+  status!: SCENARIO_STATUS;
+
+  @ApiPropertyOptional()
+  @Column({ type: 'jsonb', nullable: true })
+  metadata?: JSON;
+
+  @Column({
+    type: 'timestamp',
+    default: () => 'CURRENT_TIMESTAMP',
+  })
+  lastEdited: string;
+
+  @ManyToOne(() => User, (user: User) => user.scenarios, {
+    eager: false,
+  })
+  @ApiProperty()
+  user?: User;
+
+  /**
+   * @debt Auto-assign user and make not nullable
+   */
+  @Column({ nullable: true })
+  @ApiPropertyOptional()
+  userId?: string;
+
+  @ManyToOne(() => User, (user: User) => user.scenariosLastEdited, {
+    eager: false,
+  })
+  @ApiProperty()
```

Comment:
Again Circular Dep. isssue.

I'd fix this myself but not sure how you want to tackle it because we are following different appraches through entities